test/test_baseplanes.py

```diff
@@ -0,0 +1,99 @@
+        baseplanes.set_base_plane_dict({'x': 1, 'y': 0, 'z': 0.2, 'xy_angle': 0, 'elevation': 1}, 1)
+        baseplanes.set_base_plane_dict({'x': 5, 'y': 3, 'z': -10, 'xy_angle': 3, 'elevation': 0.5}, 2)
+        baseplanes.set_base_plane_dict({'x': -0.3, 'y': 1, 'z': 3, 'xy_angle': 0, 'elevation': -6}, 3)
+
+        self.assertEqual(baseplanes.L0_base_plane_dict, {'x': 1, 'y': 2, 'z': 0, 'xy_angle': -0.8, 'elevation': 1.3})
+        self.assertEqual(baseplanes.L1_base_plane_dict, {'x': 1, 'y': 0, 'z': 0.2, 'xy_angle': 0, 'elevation': 1})
+        self.assertEqual(baseplanes.L2_base_plane_dict, {'x': 5, 'y': 3, 'z': -10, 'xy_angle': 3, 'elevation': 0.5})
+        self.assertEqual(baseplanes.L3_base_plane_dict, {'x': -0.3, 'y': 1, 'z': 3, 'xy_angle': 0, 'elevation': -6})
+
+    def test_invalid_input(self):
+        baseplanes = BasePlane()
+
+        with self.assertRaises(ValueError):
+            baseplanes.set_base_plane_dict({'x': 1, 'y': 2, 'z': 0, 'xy_angle': -0.8, 'elevation': 1.3}, 4)
+
+
+class TestGetBasePlane(unittest.TestCase):
+
+    def test_origin(self):
+        baseplanes = BasePlane()
+
+        empty_list = [0, 0, 0]
+        self.assertEqual(empty_list, baseplanes.get_base_plane(0)[0].tolist())
+        self.assertEqual(empty_list, baseplanes.get_base_plane(1)[0].tolist())
+        self.assertEqual(empty_list, baseplanes.get_base_plane(2)[0].tolist())
+        self.assertEqual(empty_list, baseplanes.get_base_plane(3)[0].tolist())
+
+        baseplanes.set_base_plane_dict({'x': 1, 'y': 2, 'z': 0, 'xy_angle': -0.8, 'elevation': 1.3}, 0)
+        baseplanes.set_base_plane_dict({'x': 1, 'y': 0, 'z': 0.2, 'xy_angle': 0, 'elevation': 1}, 1)
+        baseplanes.set_base_plane_dict({'x': 5, 'y': 3, 'z': -10, 'xy_angle': 3, 'elevation': 0.5}, 2)
+        baseplanes.set_base_plane_dict({'x': -0.3, 'y': 1, 'z': 3, 'xy_angle': 0, 'elevation': -6}, 3)
+
+        self.assertEqual([1, 2, 0], baseplanes.get_base_plane(0)[0].tolist())
+        self.assertEqual([1, 0, 0.2], baseplanes.get_base_plane(1)[0].tolist())
+        self.assertEqual([5, 3, -10], baseplanes.get_base_plane(2)[0].tolist())
+        self.assertEqual([-0.3, 1, 3], baseplanes.get_base_plane(3)[0].tolist())
+
+    def test_normal(self):
+        baseplanes = BasePlane()
+
+        self.assertEqual([1, 0, 0], baseplanes.get_base_plane(0)[1].tolist())
+        self.assertEqual([1, 0, 0], baseplanes.get_base_plane(1)[1].tolist())
+        self.assertEqual([1, 0, 0], baseplanes.get_base_plane(2)[1].tolist())
+        self.assertEqual([1, 0, 0], baseplanes.get_base_plane(3)[1].tolist())
+
+        baseplanes.set_base_plane_dict({'x': 1, 'y': 2, 'z': 0, 'xy_angle': np.pi/2, 'elevation': 0}, 0)
+        self.assertAlmostEqual(0, baseplanes.get_base_plane(0)[1].tolist()[0])
+        self.assertAlmostEqual(1, baseplanes.get_base_plane(0)[1].tolist()[1])
+        self.assertAlmostEqual(0, baseplanes.get_base_plane(0)[1].tolist()[2])
```

Comment:
`np` has some testing helpers that you could use here:

https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_almost_equal.html

Not sure if its worth your time to refactor everything if its already working, but it may be helpful to know for future tests.

Reply:
They also have an assert equal if you are working with exact types (like `int` above)

https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_equal.html

Reply:
So I tried having those in an assertEqual statement but due to numerical "this is not exactly 0 but just almost", I changed it to assertAlmostEqual... I think I'll just leave it like this for now, refactoring this seems unnecessary. But thanks for the suggestion!